pkg/localstore/sampler.go

```diff
@@ -180,8 +188,28 @@ func (db *DB) ReserveSample(
 		} else {
 			currentMaxAddr = swarm.NewAddress(make([]byte, 32))
 		}
-		if le(item.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
-			insert(item)
+		if le(item.transformedAddress.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
+
+			chunk := swarm.NewChunk(swarm.NewAddress(item.chunkItem.Address), item.chunkItem.Data)
+
+			stamp := postage.NewStamp(item.chunkItem.BatchID, item.chunkItem.Index, item.chunkItem.Timestamp, item.chunkItem.Sig)
+
+			stampData, err := stamp.MarshalBinary()
+			if err != nil {
+				logger.Info("error marshaling stamp for chunk", chunk.Address(), err)
+				continue
+			}
+			_, err = db.validStamp(chunk, stampData)
```

Comment:
Before merging this branch, I would suggest that we test the branch on mainnet, and see how many of these errors we actually see. Maybe it's not worth adding it.

Reply:
If a node has lots of pinned chunks with expired stamps, you'll see a LOT of them.  And that node would not be able to participate in storage incentives because of those expired-but-locally-pinned chunks.

Reply:
We ran this on `dev-bee-storage` nodes. We do see some invalid stamp messages. But we dont see the `data invalid` log. So this means our assumptions are correct.

@ldeffenb With this change, it could be possible that your node could be able to participate if it can check and discard all the uninteresting chunks within the round time.

Reply:
> @ldeffenb With this change, it could be possible that your node could be able to participate if it can check and discard all the uninteresting chunks within the round time.

Once this is on the mainnet, I might do a hack to have it generate a sample even though it is not staked/not playing.   Then I can see the time it takes to generate the sample (and the system load as well), and compare my hash manually with what is eventually revealed to know if it is safe to stake.   We (@mfw78 and I) are working on a monitor similar to monBee that we'll hopefully release on Monday that makes the schelling game pretty easy to watch.

Reply:
@aloknerurkar @istae Do you know if the blocksPerRound will remain at 152 in the gnosis deployment resulting in ~3 times faster game rounds, or will it go to 364 to keep the rounds at approximately 30 minutes each?   Or maybe land somewhere in between at some other integer multiple of 4?  Just curious for planning purposes.  My testnet nodes at depth 2 are taking anywhere from 3-5 minutes to generate their samples as a point of reference.

Reply:
we intend to keep 152 in gnosis, however this might still change. the actual time window to calculate the sample and apply is the claim + commit phase, so for example you should be fine with 3-5 minute calculation